pygmt/src/grdclip.py

```diff
@@ -0,0 +1,91 @@
+"""
+grdclip - Change the range and extremes of grid values.
+"""
+
+import xarray as xr
+from pygmt.clib import Session
+from pygmt.helpers import (
+    GMTTempFile,
+    build_arg_string,
+    fmt_docstring,
+    kwargs_to_strings,
+    use_alias,
+)
+
+
+@fmt_docstring
+@use_alias(
+    G="outgrid",
+    R="region",
+    Sa="above",
+    Sb="below",
+    Si="between",
+    Sr="new",
+    V="verbose",
+)
+@kwargs_to_strings(
+    R="sequence",
+    Sa="sequence",
+    Sb="sequence",
+    Si="sequence",
+    Sr="sequence",
+)
+def grdclip(grid, **kwargs):
+    r"""
+    Sets values in a grid that meet certain criteria to a new value.
+
+    Produce a clipped ``outgrid`` or xarray.Data Array version of the
+    input ``grid`` file.
+
+    The parameters ``above`` and ``below`` allow for a given value to be set
+    for values above or below a set amount, respectively. This allows for
+    extreme values in a grid, such as points below a certain depth when
+    plotting Earth relief, to all be set to the same value.
+
+    Full option list at :gmt-docs:`grdclip.html`
+
+    {aliases}
+
+    Parameters
+    ----------
+    grid : str or xarray.DataArray
+        The file name of the input grid or the grid loaded as a DataArray.
+    outgrid : str or None
+        The name of the output netCDF file with extension .nc to store the grid
+        in.
+    {R}
+    above : str or list or tuple
+        [*high*, *above*].
+        Set all data[i] > *high* to *above*.
+    below : str or list or tuple
+        [*low*, *below*].
+        Set all data[i] < *low* to *below*.
+    between : str or list or tuple
+        [*low*, *high*, *between*].
+        Set all data[i] >= *low* and <= *high* to *between*.
+        Repeat the option for as many intervals as are needed.
```

Comment:
> Repeat the option for as many intervals as are needed.

It's impossible to repeat the option in Python.

Reply:
What if we could do it like frame (-B) in `fig.basemap` and pass in a list? Something like `between=['0/2/1', '3/5/4']`? Or a list of tuples like `between=[(0, 2 ,1), (3, 5 ,4)]` that expands to `-Sb0/2/1 -Sb3/5/4` in GMT.

Reply:
`between=[(0, 2 ,1), (3, 5 ,4)]` makes sense to me, but it needs more coding afforts.

Reply:
This is fixed in 1046419. Regarding @weiji14's suggestion, I'm assuming that would require a change to `kwargs_to_strings()`? Assuming so, I would prefer for this current method to be accepted as is, and a future pull request can be submitted once `kwargs_to_strings` is updated.

Reply:
Actually I was thinking of handling this feature by changing `Sb='sequence'`. Perhaps implementing a `sequence_tuple`? But yes, this can be done in a separate enhancement PR.

Reply:
> this can be done in a separate enhancement PR.

I agree.